naas/models/project_subscriber_profile.py

```diff
@@ -0,0 +1,114 @@
+        :param params: dict
+        :return: ProjectSubscriberProfile
+        """
+        if not params:
+            params = {}
+
+        request = naas.requests.ProjectSubscriberProfiles.retrieve_by_project_id_and_project_subscriber_id(
+            project_id, project_subscriber_id, params
+        )
+
+        if request:
+            data = request.json().get('data')
+            data.update({'project_id': project_id})
+            data.update({'project_subscriber_id': project_subscriber_id})
+            return cls(data)
+
+        Configuration(
+            {
+                "logger": ("Failure retrieving the project subscriber profile "
+                           f"{request.status_code}")
+            }
+        )
+
+    @classmethod
+    def update_by_project_id_and_project_subscriber_id(
+            cls, project_id, project_subscriber_id, params=None):
+        """
+        Update an existing project subscriber profile
+        :param project_id: str
+        :param project_subscriber_id: str
+        :param params: dict
+        :return: ProjectSubscriberProfile
+        """
+        if not params:
+            params = {}
+
+        request = naas.requests.ProjectSubscriberProfiles.update_by_project_id_and_project_subscriber_id(
+            project_id, project_subscriber_id, params
+        )
+
+        if request:
+            data = request.json().get('data')
+            data.update({'project_id': project_id})
+            data.update({'project_subscriber_id': project_subscriber_id})
+            return cls(data)
+
+        error = Error(request.json().get('data'))
+        failure_message = (
+            f"Failure creating the record {error.full_messages()}")
+        Configuration({"logger": f"{failure_message}"})
+        raise InvalidRequestError(failure_message)
+
+    def project_id(self):
+        """Returns the associated project id"""
+        return self.attributes.get('project_id')
+
+    def project_attributes(self):
+        """Returns the project attributes"""
+        return self.attributes.get('project', {})
+
+    def project(self):
+        """Returns an instance of the project"""
+        if self.project_attributes():
+            return Project(self.project_attributes())
+        return Projects.retrieve(self.project_id())
+
+    def project_subscriber_id(self):
+        """Returns the project subscriber id"""
+        return self.attributes.get('project_subscriber_id')
+
+    def project_subscriber_attributes(self):
+        """Returns the project subscriber attributes"""
+        return self.attributes.get('project_subscriber', {})
+
+    def project_subscriber(self):
+        """Returns an instance of the project subscriber"""
+        if self.project_subscriber_attributes():
+            return ProjectSubscriber(self.project_subscriber_attributes())
+        return ProjectSubscribers.retrieve_by_project_id(
+            self.project_id(), self.project_subscriber_id())
```

Comment:
This throws an error because there is no `project_id()` (str), it's `project_id` because it gets set this way in the initializer.